dotCMS/src/main/java/com/dotmarketing/startup/AbstractJDBCStartupTask.java

```diff
@@ -98,48 +174,570 @@ public boolean equals(Object obj) {
+			return String.format(customQuery, constraint.tableName, constraint.name, constraint.value,
+					constraint.columnName);
+		}
+
+	}
+
+	/**
+	 * Represents the check constraint that needs to be added in a special way.
+	 * 
+	 * @author Jose Castro
+	 * @version 3.7
+	 * @since Dec 12, 2016
+	 *
+	 */
+	public class CheckConstHandler implements ConstraintHandler<Constraint> {
+
+		final private String constraintName;
+		final private String customQuery;
+
+		/**
+		 * Creates a handler for creating check constraint objects. In order to
+		 * allow more flexibility to the process, this constraint name can also
+		 * be the initial sequence of characters that make up the constraint.
+		 * This way, you can run the same custom SQL query for more than one
+		 * constraint. This is particularly useful when running a SQL query for
+		 * constraints that are generated automatically by a given database.
+		 * 
+		 * @param constraintName
+		 *            - The name of the constraint (or the beginning of it) that
+		 *            needs to be created in a special way.
+		 * @param customQuery
+		 *            - The SQL statement that will be run. It might contain
+		 *            placeholders to replace with the actual query data.
+		 */
+		public CheckConstHandler(final String constraintName, final String customQuery) {
+			this.constraintName = constraintName;
+			this.customQuery = customQuery;
+		}
+
+		@Override
+		public String getConstraintName() {
+			return constraintName;
+		}
+
+		@Override
+		public String generateQuery(final Constraint constraint) {
+			return String.format(customQuery, constraint.tableName, constraint.name, constraint.value);
+		}
+
+	}
+
+	@Override
 	public void executeUpgrade() throws DotDataException, DotRuntimeException{
 		DotConnect dc = new DotConnect();
 		Connection conn = null;
 		List<PrimaryKey> primaryKeys=null;
 		List<ForeignKey> foreignKeys=null;
 		List<Index> indexes=null;
-		
+		List<Constraint> defaultConstraints = null;
+		List<Constraint> checkConstraints = null;
 		try {
+			this.primaryKeyProcessor = getPrimaryKeyProcessor();
+			this.foreignKeyProcessor = getForeignKeyProcessor();
+			this.indexProcessor = getIndexProcessor();
+			this.defaultConstProcessor = getDefaultConstraintProcessor();
+			this.checkConstProcessor = getCheckConstraintProcessor();
 		    conn = DbConnectionFactory.getDataSource().getConnection();
 			conn.setAutoCommit(true);
-			
 			List<String> tables = getTablesToDropConstraints();
 			if(tables!=null){
-                foreignKeys=getForeingKeys(conn, tables,true);
-                //conn.commit();
-                primaryKeys=getPrimaryKey(conn, tables, true);
-                //conn.commit();
-                indexes=getIndexes(conn, tables,true);
-                //conn.commit();
-                if(DbConnectionFactory.isMsSql())
+				boolean executeDrop = true;
+				logTaskProgress("==> Retrieving foreign keys [Drop objects? " + executeDrop + "]");
```

Comment:
@jcastro-dotcms I have one question, we are dropping constraints, keys, etc, etc when we found tables using **getTablesToDropConstraints** but what about having tables in **getTablesToDropConstraints** and for example the **getPostgresScript** method returning null?.

I ask because this is a **MSSQL** only upgrade task and looks like that event if the **getPostgresScript** method returns null we are going to do all those drops for that db.

Looks like we should only drop if **getTablesToDropConstraints()** and **getXXXScript()** != null